premium/usage.go

```diff
@@ -41,6 +45,11 @@ const (
 	MaximumUpdateIntervalHeader = "x-cq-maximum-update-interval"
 )
 
+//go:generate mockgen -package=mocks -destination=../premium/mocks/marketplacemetering.go -source=usage.go AWSMarketplaceClientInterface
+type AWSMarketplaceClientInterface interface {
+	MeterUsage(ctx context.Context, params *marketplacemetering.MeterUsageInput, optFns ...func(*marketplacemetering.Options)) (*marketplacemetering.MeterUsageOutput, error)
+}
```

Comment:
This was added to ensure that we could mock the AWS API calls like we do the API Server cakks